packages/@react-spectrum/menu/src/ContextualHelpTrigger.tsx

```diff
@@ -58,62 +64,96 @@ function ContextualHelpTrigger(props: InternalMenuDialogTriggerProps): ReactElem
   let [trigger] = React.Children.toArray(props.children);
   let [, content] = props.children as [ReactElement, ReactElement];
 
-  let onExit = () => {
-    // if focus was already moved back to a menu item, don't need to do anything
-    if (!menu.current.contains(document.activeElement)) {
-      // need to return focus to the trigger because hitting Esc causes focus to go to the subdialog, which is then unmounted
-      // this leads to blur never being fired nor focus on the body
-      triggerRef.current.focus();
-    }
-  };
   let onBlurWithin = (e) => {
-    if (e.relatedTarget && popoverRef.current && !popoverRef.current?.UNSAFE_getDOMNode().contains(e.relatedTarget)) {
-      if (menuState.expandedKeys.has(props.targetKey)) {
-        menuState.toggleKey(props.targetKey);
+    if (e.relatedTarget && popoverRef.current && (!popoverRef?.current?.UNSAFE_getDOMNode()?.contains(e.relatedTarget) && !(e.relatedTarget === triggerRef.current && getInteractionModality() === 'pointer'))) {
+      if (submenuTriggerState.isOpen) {
+        submenuTriggerState.close();
       }
     }
   };
+
+  let overlay;
+  let tray;
+  let onBackButtonPress = () => {
+    submenuTriggerState.close();
+    if (parentMenuRef.current && !parentMenuRef.current.contains(document.activeElement)) {
+      // Delay for the parent menu in the tray to no longer be display: none so focus can properly be moved to it
+      requestAnimationFrame(() => parentMenuRef.current.focus());
+    }
+  };
+
+  if (isMobile) {
+    delete submenuTriggerProps.onBlur;
+    delete submenuTriggerProps.onHoverChange;
```

Comment:
yep, we don't want to open ContextualHelpTrigger/SubMenuTrigger on hover in a tray experience. This would only happen if you have a device with mouse support + a very small screen so pretty edge casey but can be reproduced via Chrome dev tools + device emulation + desktop mode turned on.

Reply:
wonder if we should move this into the hook at some point so others can make a tray version too

Reply:
happy to do so, but isn't our definition of mobile pretty Spectrum specific? Would we need to add a prop for users to then define what they consider 'isMobile'?